src/storage/src/render/sources.rs

```diff
@@ -602,6 +612,23 @@ where
     }
 }
 
+fn handle_missing_keys(key_arity: Option<usize>) -> Box<dyn Fn(KV) -> KV> {
```

Comment:
just make this is a normal function that takes in `key_arity` and `KV`, and return a `KV`

then, above you can do:
```
.map(|kv| handle_missing_keys_fn(key_arity, kv))
```
and rustc will setup the closure that borrows `key_arity` for you!

Reply:
oh yeah the round-about inner function was a way to create the key slice once for a given arity, so it can be more cheaply cloned for each KV. let me play around with your idea for `raise_key_value_errors` and see how it shakes out

Reply:
i think its gonna be the same price if you call clone or `pack_slice`!